capstone/src/main/java/com/google/step/servlets/RestaurantServlet.java

```diff
@@ -119,45 +101,37 @@ private String changeNonNumericCharacters(String num) {
    * @param response All details for the requested restaurant, in the following json format:
     {
       "restaurant" : {
-        "restaurantKey": <long>,
-        "name": <String>,
-        "location": <GeoPt>,
-        "story": <String>,
-        "cuisine": <List<String>>,
-        "phone": <String>,
-        "website": <String>,
-        "status": <String>
+        "restaurantKey": long,
+        "name": String,
+        "location": GeoPt,
+        "story": String,
+        "cuisine": List<String>,
+        "phone": String,
+        "website": String,
+        "status": String
       }
     }
    * @param request Specifies restaurant key, in the following format:
-     /restaurant?restaurantKey=<int>
-   * @throws SC_NOT_FOUND if no restaurant with the requested key is found in Datastore.
+     /restaurant?restaurantKey=int
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Long restaurantKeyParam = Long.parseLong(request.getParameter("restaurantKey"));
-    Query query = new Query("RestaurantInfo")
-                      .setFilter(new Query.FilterPredicate(
-                          "restaurantKey", Query.FilterOperator.EQUAL, restaurantKeyParam));
-    PreparedQuery results = datastore.prepare(query);
-    Entity resultEntity = results.asSingleEntity();
-    if (resultEntity == null) {
-      response.setStatus(HttpServletResponse.SC_NOT_FOUND);
-      return;
-    }
+    long restaurantKeyParam = Long.parseLong(request.getParameter("restaurantKey"));
 
     // Restaurant object to hold all info
-    Restaurant restaurant = Restaurant.fromEntity(resultEntity);
+    Optional<Restaurant> restaurant = restaurantClient.getSingleRestaurant(restaurantKeyParam);
 
-    // Format restaurant List to JSON for return
-    Gson gson = new Gson();
-    String json = gson.toJson(restaurant);
+    if (restaurant.isPresent()) {
+      // Format restaurant List to JSON for return
+      Gson gson = new Gson();
+      String json = gson.toJson(restaurant);
 
-    // Send the JSON as the response
-    response.setContentType("application/json;");
+      // Send the JSON as the response
+      response.setContentType("application/json;");
 
-    JsonObject ret = new JsonObject();
-    ret.addProperty("restaurant", json);
-    response.getWriter().println(ret);
+      JsonObject ret = new JsonObject();
+      ret.addProperty("restaurant", json);
+      response.getWriter().println(ret);
+    }
```

Comment:
Also consider adding a test for this case

Reply:
Yes thank you, I returned the error now!
Is it okay if I do the test case in another pull request with multiple test cases for the RestaurantServlet, since I'll probably make a new file for it and use the HTTP testing for servlets?

Reply:
Sure